internal/controller/nodeagent_test.go

```diff
@@ -1352,6 +1444,219 @@ func TestDPAReconciler_buildNodeAgentDaemonset(t *testing.T) {
+					BlockOwnerDeletion: ptr.To(true),
+				},
+			},
+		},
+		Data: data,
+	}
+}
+func TestDPAReconciler_updateNodeAgentCM(t *testing.T) {
+	testCmNs := "test-configmap-ns"
+	testCmName := "test-dpa-configmap-cm"
+	tests := []struct {
+		name                   string
+		nodeAgentConfigMap     *corev1.ConfigMap
+		dpa                    *oadpv1alpha1.DataProtectionApplication
+		wantErr                bool
+		wantNodeAgentConfigMap *corev1.ConfigMap
+	}{
+		{
+			name: "Given DPA CR instance, appropriate NodeAgent config cm is created with NodeSelector",
+			nodeAgentConfigMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      common.NodeAgentConfigMapPrefix + testCmName,
+					Namespace: testCmNs,
+				},
+			},
+			dpa: &oadpv1alpha1.DataProtectionApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testCmName,
+					Namespace: testCmNs,
+				},
+				Spec: oadpv1alpha1.DataProtectionApplicationSpec{
+					Configuration: &oadpv1alpha1.ApplicationConfig{
+						NodeAgent: &oadpv1alpha1.NodeAgentConfig{
+							NodeAgentCommonFields: oadpv1alpha1.NodeAgentCommonFields{
+								PodConfig: &oadpv1alpha1.PodConfig{
+									NodeSelector: map[string]string{"foos": "bars"},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+			wantNodeAgentConfigMap: createTestBuiltNodeAgentCM(map[string]string{
+				"node-agent-config": `{
+					"loadAffinity": [
+						{
+							"nodeSelector": {
+								"foos": "bars"
+							}
+						}
+					]
+				}`,
+			}),
+		},
+		{
+			name: "Given DPA CR instance, appropriate NodeAgent config cm is created with LoadConcurrency, BackupPVCConfig, RestorePVCConfig, PodResources",
+			nodeAgentConfigMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      common.NodeAgentConfigMapPrefix + testCmName,
+					Namespace: testCmNs,
+				},
+			},
+			dpa: &oadpv1alpha1.DataProtectionApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testCmName,
+					Namespace: testCmNs,
+				},
+				Spec: oadpv1alpha1.DataProtectionApplicationSpec{
+					Configuration: &oadpv1alpha1.ApplicationConfig{
+						Velero: &oadpv1alpha1.VeleroConfig{},
+						NodeAgent: &oadpv1alpha1.NodeAgentConfig{
+							NodeAgentCommonFields: oadpv1alpha1.NodeAgentCommonFields{},
+							NodeAgentConfigMapSettings: oadpv1alpha1.NodeAgentConfigMapSettings{
+								LoadConcurrency: &oadpv1alpha1.LoadConcurrency{
+									GlobalConfig: 10,
+									PerNodeConfig: []oadpv1alpha1.RuledConfigs{
+										{
+											NodeSelector: metav1.LabelSelector{
+												MatchLabels: map[string]string{"app": "velero"},
+											},
+											Number: 1,
+										},
+										{
+											NodeSelector: metav1.LabelSelector{
+												MatchLabels: map[string]string{"app": "velero-2"},
+											},
+											Number: 5,
+										},
+									},
+								},
+								BackupPVCConfig: map[string]nodeagent.BackupPVC{
+									"pvc-1": {
```

Comment:
@mpryc This map key isn't the PVC name, its a storageClass name. 
I see that this is what got @weshayutin confused.

Reply:
well to be fair, I'm easily confused.  I DO think we'll have to be careful w/ this one when handing it off to QE :)